src/api/firebase.ts

```diff
@@ -266,10 +267,17 @@ export async function updateItem(listPath: string, item: ListItem) {
 	}
 }
 
-export async function deleteItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to delete an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+//delete an item from the list
+export async function deleteItem(listPath: string, item: ListItem) {
+	const itemDocRef = doc(db, listPath, "items", item.id);
```

Comment:
Great point! We try to use nullable types ass much as possible!
![image](https://github.com/user-attachments/assets/c3270424-7a90-4dbb-9b63-b6386e517830)

Reply:
We were able to do a little razzle dazzle like this!
![image](https://github.com/user-attachments/assets/c97336c4-1edd-44d8-a42b-e33ba7480207)

Reply:
@RossaMania @adidalal This seems helpful :) We can play around with firebase getting documents methods 
<img width="616" alt="Screenshot 2024-09-21 at 9 59 13 PM" src="https://github.com/user-attachments/assets/93cd65e9-fb0b-4d83-870d-856e1ae7bd23">

Reply:
I see what you are saying now! The problem could be that FireBase is not getting the item to begin with. So we need to make sure that the item document is grabbed from firebase properly.

Reply:
@eternalmaha brought up, since this is the file and functions talking to the firebase, we should use the .exists() method! I haven't seen that since MongoDB!
![image](https://github.com/user-attachments/assets/1e7cacaf-b35d-4903-8fc6-f2c74f29526d)